src/main/java/com/programmers/ticketparis/domain/reservation/Reservation.java

```diff
@@ -39,7 +40,7 @@ private Reservation(ReservationStatus reservationStatus, Long customerId, Long s
 
     private ReservationStatus checkReservationStatus(ReservationStatus reservationStatus) {
         if (reservationStatus != ReservationStatus.COMPLETED) {
-            throw new ReservationException(NOT_EXIST_RESERVATION_STATUS, List.of(reservationStatus.toString()));
+            throw new ReservationException(RESERVATION_STATUS_INVALID, List.of(reservationStatus.toString()));
```

Comment:
`p3`

현재 BusinessException의 생성자 파라미터가 아래와 같아서 예외를 생성할때마다 원인 파라미터 정보를 List로 감싸주고 있는데요,

```java
@Getter
public class BusinessException extends RuntimeException {

    private final ExceptionRule rule;
    private final List<String> rejectedValues;

    protected BusinessException(ExceptionRule rule, List<String> rejectedValues) {
        super(rule.getMessage());
        this.rule = rule;
        this.rejectedValues = rejectedValues;
    }
}
```

여기서 파라미터를 가변인자로 바꿔주면 사용되는 쪽에서 List로 감싸줄 필요가 없을 것 같아요~

```java
@Getter
public class BusinessException extends RuntimeException {

    private final ExceptionRule rule;
    private final String[] rejectedValues;

    protected BusinessException(ExceptionRule rule, String... rejectedValues) {
        super(rule.getMessage());
        this.rule = rule;
        this.rejectedValues = rejectedValues;
    }
}
```

이렇게 사용하면 아래처럼 바꿀 수 있겠죠?

```suggestion
            throw new ReservationException(RESERVATION_STATUS_INVALID, reservationStatus.toString());
```